editor.planx.uk/src/components/Feedback/FeedbackForm.test.tsx

```diff
@@ -0,0 +1,147 @@
+
+const mockUnlabeledInput: FeedbackFormInput[] = [
+  {
+    name: "userComment",
+    ariaDescribedBy: "comment-title",
+  },
+];
+
+const mockLabeledInputs: FeedbackFormInput[] = [
+  { name: "userContext", id: "userContext", label: "User Context" },
+  { name: "userComment", id: "userComment", label: "User Comment" },
+];
+
+describe("FeedbackForm functionality", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("renders unlabelled input correctly", () => {
+    const { getByLabelText } = setup(
+      <FeedbackForm
+        inputs={mockUnlabeledInput}
+        handleSubmit={mockHandleSubmit}
+      />,
+    );
+    const renderedInput = getByLabelText("Leave your feedback");
+    expect(renderedInput).toBeInTheDocument();
+  });
+
+  test("renders labelled inputs correctly", () => {
+    const { getByLabelText } = setup(
+      <FeedbackForm
+        inputs={mockLabeledInputs}
+        handleSubmit={mockHandleSubmit}
+      />,
+    );
+    const contextInput = getByLabelText("User Context");
+    const commentInput = getByLabelText("User Comment");
+    expect(contextInput).toBeInTheDocument();
+    expect(commentInput).toBeInTheDocument();
+  });
+
+  test("renders the feedback disclaimer with the input", () => {
+    const { getByText } = setup(
+      <FeedbackForm
+        inputs={mockUnlabeledInput}
+        handleSubmit={mockHandleSubmit}
+      />,
+    );
+    expect(
+      getByText(
+        /Do not share personal or financial information in your feedback./i,
+      ),
+    ).toBeInTheDocument();
+  });
+
+  test("can submit an unlabelled input", async () => {
+    const { getByLabelText, getByText } = setup(
+      <FeedbackForm
+        inputs={mockUnlabeledInput}
+        handleSubmit={mockHandleSubmit}
+      />,
+    );
+    await userEvent.type(
+      getByLabelText("Leave your feedback"),
+      "This is a test comment",
+    );
+    await userEvent.click(getByText("Send feedback"));
+
+    await waitFor(() => {
+      expect(mockHandleSubmit).toHaveBeenCalledWith(
+        expect.objectContaining({
+          userComment: "This is a test comment",
+        }),
+        expect.any(Object),
+      );
+    });
+  });
+
+  test("can submit labelled inputs", async () => {
+    const { getByLabelText, getByText } = setup(
+      <FeedbackForm
+        inputs={mockLabeledInputs}
+        handleSubmit={mockHandleSubmit}
+      />,
+    );
+
+    await userEvent.type(
+      getByLabelText("User Context"),
+      "This is test context",
+    );
+    await userEvent.type(
+      getByLabelText("User Comment"),
+      "This is a test comment",
+    );
+
+    await userEvent.click(getByText("Send feedback"));
+
+    await waitFor(() => {
+      expect(mockHandleSubmit).toHaveBeenCalledWith(
+        expect.objectContaining({
+          userContext: "This is test context",
+          userComment: "This is a test comment",
+        }),
+        expect.any(Object),
+      );
+    });
+  });
+});
```

Comment:
_Comment:_

I was getting some issues that the expectation was failing because Formik passes both the `values` and also helper functions etc as two arguments to the submit function. 

I can't seem to see why this would be an issue based on docs / other tests but I got a workaround by splitting the arguments into separate expectations.